src/inference/src/dev/core_impl.cpp

```diff
@@ -359,26 +353,27 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::string& mod
     auto parsed = parseDeviceNameIntoConfig(device_name, config);
     auto plugin = get_plugin(parsed._deviceName);
     ov::SoPtr<ov::ICompiledModel> res;
-    auto cacheManager =
-        coreConfig.get_cache_config_for_device(parsed._deviceName, device_supports_cache_dir(plugin), parsed._config)
-            ._cacheManager;
+    property_manager.merge_property(parsed._config, plugin.get_name());
```

Comment:
changing global state here => not allowed

Reply:
If not changing global state, we will have to create another cache for the core properties item of `config` during compile_model(), there will be 3 level cache as priority from low to high:
```
     1.  global default core properties  -  from core.set_property()
     2.  plugin default core properties  - from core.set_property(device_name,)
     3.  plugin core properties   - from core.compile_model(,) 
```
How do you think about it?